ocaml/xapi/xapi_network.ml

```diff
@@ -162,31 +184,43 @@ let counter = ref 0
 let mutex = Mutex.create ()
 let stem = "xapi"
 
-let pool_introduce ~__context ~name_label ~name_description ~mTU ~other_config ~bridge =
+let pool_introduce ~__context ~name_label ~name_description ~mTU ~other_config ~bridge ~managed =
   let r = Ref.make() and uuid = Uuid.make_uuid() in
   Db.Network.create ~__context ~ref:r ~uuid:(Uuid.to_string uuid)
     ~current_operations:[] ~allowed_operations:[]
-    ~name_label ~name_description ~mTU ~bridge
+    ~name_label ~name_description ~mTU ~bridge ~managed
     ~other_config ~blobs:[] ~tags:[] ~default_locking_mode:`unlocked ~assigned_ips:[];
   r
+let choose_bridge_name ~__context ~bridge ~bridges =
+  if bridge = "" then
+    let rec loop () =
+      let name = stem ^ (string_of_int !counter) in
+      incr counter;
+      if List.mem name bridges then loop ()
+      else
+        name in
+    loop ()
+  else
+    bridge
 
-let create ~__context ~name_label ~name_description ~mTU ~other_config ~tags =
+let create ~__context ~name_label ~name_description ~mTU ~other_config ~bridge ~managed ~tags =
   Mutex.execute mutex (fun () ->
       let networks = Db.Network.get_all ~__context in
       let bridges = List.map (fun self -> Db.Network.get_bridge ~__context ~self) networks in
       let mTU = if mTU <= 0L then 1500L else mTU in
-      let rec loop () =
-        let name = stem ^ (string_of_int !counter) in
-        incr counter;
-        if List.mem name bridges then loop ()
-        else
-          let r = Ref.make () and uuid = Uuid.make_uuid () in
-          Db.Network.create ~__context ~ref:r ~uuid:(Uuid.to_string uuid)
-            ~current_operations:[] ~allowed_operations:[]
-            ~name_label ~name_description ~mTU ~bridge:name
-            ~other_config ~blobs:[] ~tags ~default_locking_mode:`unlocked ~assigned_ips:[];
-          r in
-      loop ())
+      if String.length bridge > 15 then
```

Comment:
I think it would be clearer to have a block to define the bridge name here, like
```
let bridge =
  if bridge = "" then
    choose_bridge_name bridges
  else begin
    ...assertions...;
    bridge
  end
in
```
So `choose_bridge_name` just chooses a new bridge name. And in the `Db.Network.create` line we just have `~bridge`.